include/logging/log_core.h

```diff
@@ -185,6 +185,49 @@ extern "C" {
 		}							 \
 	} while (false)
 
+static inline log_arg_t __do_strdup(u32_t msk, u32_t idx, log_arg_t param)
+{
+	if (msk & (1<<idx)) {
+		const char *str = (const char *)param;
+		char *log_strdup(const char *str);
+
+		param = (log_arg_t)log_strdup(str);
```

Comment:
`log_generic` verifies `if (!log_is_strdup(str)`, shall we check this here too?

Reply:
If strdup is not allowed in OT, we don't have to check it.
But agree, log_is_strdup check shall be optional in this macro.